activesupport/lib/active_support/deprecation/instance_delegator.rb

```diff
@@ -18,7 +17,27 @@ def include(included_module)
         end
 
         def method_added(method_name)
-          singleton_class.delegate(method_name, to: :instance)
+          use_instead =
+            case method_name
+            when :silence, :behavior=, :disallowed_behavior=, :disallowed_warnings=, :silenced=, :debug=
+              "Rails.application.deprecators.#{method_name}"
+            when :warn, :deprecate_methods, :gem_name, :gem_name=, :deprecation_horizon, :deprecation_horizon=
+              "your own Deprecation object"
+            else
+              "Rails.application.deprecators[framework].#{method_name} where framework is for example :active_record"
+            end
+          args = /[^\]]=\z/.match?(method_name) ? "arg" : "..."
+          singleton_class.module_eval <<~RUBY, __FILE__, __LINE__ + 1
+            def #{method_name}(#{args})
+              ActiveSupport.deprecator.warn("Calling #{method_name} on ActiveSupport::Deprecation is deprecated and will be removed from Rails (use #{use_instead} instead)")
+              ActiveSupport::Deprecation._instance.#{method_name}(#{args})
```

Comment:
I think silence, behavior, etc should be redirected to `Rails.application.deprecators`, not `_instance`... otherwise they're going to immediately stop affecting all Rails-generated deprecations that have already been pointed elsewhere.

These lines also need to go the other way around, at minimum, to have some hope of directing the meta-deprecation warning towards the intended deprecation config.

I just discovered that my application had been failing to identify deprecation warnings for a while because `ActiveSupport::Deprecation.disallowed_warnings = :all` was no longer having the intended effect.

Reply:
Well, this PR only keeps the current main behavior and only adds the deprecation for calling `ActiveSupport::Deprecation.disallowed_warnings=`, but there is indeed a loss of behavior between `ActiveSupport::Deprecation.disallowed_warnings=` in 7.0 and here.

I do wonder how much we want to support this "undocumented behavior" though. For example https://api.rubyonrails.org/v7.0/classes/ActiveSupport/Deprecation/Disallowed.html mentions `ActiveSupport::Deprecation.disallowed_behavior`, but documents an instance writer for `disallowed_warnings`, which means it should have been used as `ActiveSupport::Deprecation.new.disallowed_warnings=`, or maybe `ActiveSupport::Deprecation.instance.disallowed_warnings=` (since `Singleton` was documented as an included module).

But the "correct" way was to set the configuration on `Rails.application.config.active_support`, and that has preserved the correct behavior in https://github.com/rails/rails/pull/46328.

There was a mention in the [Configuring guide](https://guides.rubyonrails.org/v7.0/configuring.html#config-active-support-disallowed-deprecation-warnings) to using the instance delegators as an alternative so we can definitely argue it was documented behavior.
> Alternatively, you can set `ActiveSupport::Deprecation.disallowed_warnings`

Anyway, I'll try to do the right thing and restore the 7.0 behavior of not only impacting the singleton instance but all deprecators when the instance delegators are used for these settings. The deprecation message correctly points how to get the previous behavior without a deprecation by setting on `Rails.application.deprecators`, so we should do the same.

Reply:
> I just discovered that my application had been failing to identify deprecation warnings for a while because `ActiveSupport::Deprecation.disallowed_warnings = :all` was no longer having the intended effect.

My apologies! :pray:  My (loose) reasoning was:

1. Previous deprecations that exist in 7.0 will (likely) be removed for 7.1.
2. New deprecations that have been added for 7.1 will require new `silence` blocks, `disallowed_warnings` values, etc for applications.
3. Since applications will already be updating `silence` blocks, `disallowed_warnings` values, etc to match the new 7.1 deprecations, it would be reasonable for them to switch from `ActiveSupport::Deprecation.*` to `Rails.application.deprecators.*` at the same time.

I did not take into account applications that are between 7.0 and 7.1 (i.e tracking `main` branch).

Reply:
> Since applications will already be updating silence blocks, disallowed_warnings values, etc to match the new 7.1 deprecations

Even ignoring the running-on-main state, any upgrading-to-7.1 application is potentially currently configured in some way to direct/disallow deprecations. At absolute minimum, these meta-deprecations need to go into exactly that channel. 

The whole point of our deprecation system is that we _keep things functioning as they were before_, while warning that the application should be adapted to the new way. Creating a new parallel system next to where the old one was, and then reasoning that people will start using it because it's there, does not achieve that.

> New deprecations that have been added for 7.1 will require new silence blocks, disallowed_warnings values, etc for applications.

Specific disallows perhaps, but `:all` is pretty universal.

(I also think that we should still have a way to globally silence deprecations, or to globally disallow deprecations, without involving a railties dependency, but whatever.)